pkg/operation/botanist/garbage_collection.go

```diff
@@ -67,30 +77,51 @@ func (b *Botanist) PerformGarbageCollectionSeed() error {
 // PerformGarbageCollectionShoot performs garbage collection in the kube-system namespace in the Shoot
 // cluster, i.e., it deletes evicted pods (mitigation for https://github.com/kubernetes/kubernetes/issues/55051).
 func (b *Botanist) PerformGarbageCollectionShoot() error {
-	return b.deleteEvictedPods(b.K8sShootClient, metav1.NamespaceSystem)
-}
-
-// deleteEvictedPods determine pods in state 'Evicted' in a given namespace and delete them.
-func (b *Botanist) deleteEvictedPods(client kubernetes.Client, namespace string) error {
-	podList, err := client.ListPods(namespace, metav1.ListOptions{})
+	podList, err := b.K8sShootClient.ListPods(metav1.NamespaceSystem, metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
+
+	if err := b.deleteEvictedPods(b.K8sShootClient, podList); err != nil {
+		return err
+	}
+	return b.deleteStuckTerminatingPods(b.K8sShootClient, podList)
+}
+
+// deleteEvictedPods determines pods in state 'Evicted' in a given namespace and deletes them.
+func (b *Botanist) deleteEvictedPods(client kubernetes.Client, podList *corev1.PodList) error {
 	for _, pod := range podList.Items {
-		var (
-			name   = pod.ObjectMeta.Name
-			reason = pod.Status.Reason
-		)
-		if reason != "" && strings.Contains(reason, "Evicted") {
-			b.Logger.Debugf("Deleting pod %s as its reason is %s.", name, reason)
-			err := client.DeletePod(namespace, name)
-			if apierrors.IsNotFound(err) {
-				continue
-			}
-			if err != nil {
+		if pod.Status.Reason != "" && strings.Contains(pod.Status.Reason, "Evicted") {
```

Comment:
Why not just `strings.Contains(pod.Status.Reason, "Evicted")`?

Reply:
Done